kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/ArrayResponseType.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2010-2024. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.axonframework.extensions.kotlin.serialization
+
+import org.axonframework.common.ReflectionUtils
+import org.axonframework.common.TypeReflectionUtils
+import org.axonframework.messaging.responsetypes.AbstractResponseType
+import org.axonframework.messaging.responsetypes.InstanceResponseType
+import org.axonframework.messaging.responsetypes.ResponseType
+import java.lang.reflect.Type
+import java.util.concurrent.Future
+
+/**
+ * A [ResponseType] implementation that will match with query handlers which return a multiple instances of the
+ * expected response type. If matching succeeds, the [ResponseType.convert] function will be called, which
+ * will cast the query handler it's response to an [Array] with element type [E].
+ *
+ * @param E The element type which will be matched against and converted to
+ * @author Gerard de Leeuw
+ * @see org.axonframework.messaging.responsetypes.MultipleInstancesResponseType
+ */
+class ArrayResponseType<E>(elementType: Class<E>) : AbstractResponseType<Array<E>>(elementType) {
```

Comment:
@lion7, can you explain to me what the intent is of the `ArrayResponseType` for the `KotlinSerializer` exactly?